compare_graph.py

```diff
@@ -0,0 +1,68 @@
+import pypsa
+import networkx as nx
+import matplotlib.pyplot as plt
+
+
+def graph_properties(network: pypsa.Network) -> dict:
+    """
+    Extracts the properties of the network graph.
+
+    Parameters: 
+    network: pypsa.Network
+    Returns:
+    dict: network_properties
+    """
+
+    # extract graph properties
+    graph = network.graph()
+    network_degree = dict(graph.degree())
+    network_average_degree_connectivity = nx.average_degree_connectivity(graph)
+    network_edges = list(graph.edges())
+    network_nodes = list(graph.nodes())
+    network_number_of_edges = graph.number_of_edges()
+    network_number_of_nodes = graph.number_of_nodes()
+
+    network_properties = {
+        "network_edges": network_edges,
+        "network_degree": network_degree,
+        "network_average_degree_connectivity": network_average_degree_connectivity,
+        "network_nodes": network_nodes,
+        "network_number_of_edges": network_number_of_edges,
+        "network_number_of_nodes": network_number_of_nodes,
+    }
+
+    return network_properties
+
+
+def plot_network_graph(network: pypsa.Network, seed: int = 1969):
+    """
+    Plots the network graph.
+
+    Parameters: 
+        network: pypsa.Network
+    Returns:
+        None
+    """
+    graph = network.graph()
+    options = {"node_color": "black", "node_size": 50,
+               "linewidths": 0, "width": 0.1}
+    pos = nx.spring_layout(graph, seed=seed)  # Seed for reproducible layout
+    nx.draw(graph, pos, **options)
+    plt.savefig("network_graph.png", dpi=300)
+
+
+if __name__ == "__main__":
```

Comment:
the above code looks great!
Let's leave the code below out as its purpose is to experiment with the above code - for now let's keep experimentation local, we can set up some separate experimentation scripts at a later date

Reply:
I think we can still leave the experimentation, as it only runs when you run the script alone. If you import the script as a module, the experimentation wont run.

Reply:
Depending on what we later agree on. I will just comment it out for now.

Reply:
the repo is for doing a general network comparison so shouldn't involve any specific files. It's also not good practice to have commented out code in a repository

Reply:
feel free to keep a copy of the experimentation code as we can use it in our experimentation files